packages/react-native-aztec/ios/RNTAztecView/RCTAztecView.swift

```diff
@@ -355,16 +349,18 @@ class RCTAztecView: Aztec.TextView {
     }
 
     // MARK: - Dictation
-
-    override func dictationRecordingDidEnd() {
-        isInsertingDictationResult = true
-    }
-
-    public override func insertDictationResult(_ dictationResult: [UIDictationPhrase]) {
-        let objectPlaceholder = "\u{FFFC}"
-        let dictationText = dictationResult.reduce("") { $0 + $1.text }
-        isInsertingDictationResult = false
-        self.text = self.text?.replacingOccurrences(of: objectPlaceholder, with: dictationText)
+    
+    func removeUnicodeAndRestoreCursor(from textView: UITextView) {
+        // Capture current cursor position
+        let originalPosition = textView.offset(from: textView.beginningOfDocument, to: textView.selectedTextRange?.start ?? textView.beginningOfDocument)
```

Comment:
When running test 5 I seemed to encounter an off-by-one where the cursor advanced its position when I disabled dictation. I'm not super familiar with the textView cursor position logic, but I wonder if when we calculate the offset under the condition that `textView.selectedTextRange?.start` isn't nil, and subtracted one from the resulting integer, if that would resolve the problem. 

Any thoughts on that @SiobhyB? I'm unsure of the pitfalls of that logic the integer is `0`. IMO we shouldn't let it hold up this PR, so I'm OK with noting it for a subsequent fix.